### PR TITLE
T4857: snmp: Fix error when not defining client|network under community

### PR DESCRIPTION
**Pull request by nicolas-fort**
## Change Summary

After including FRR recommendations on snmp, we get error when not defining client or network under community.
Error reported in this [blog post](https://forum.vyos.io/t/unable-to-query-snmp-anymore-in-a-more-recent-1-4-version/10388).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4857

## Component(s) name

snmp

## How to test

```
## Simple config, with no client/network defined
vyos@vyos:~$ show config comm | grep snmp
set service snmp community FOO authorization 'ro'
set service snmp contact 'EMAIL'
set service snmp location 'LOCATION'

## Check service is running (before this patch an error was noted here)
vyos@vyos:~$ sudo systemctl status snmpd
● snmpd.service - Simple Network Management Protocol (SNMP) Daemon.
     Loaded: loaded (/lib/systemd/system/snmpd.service; disabled; vendor preset: enabled)
    Drop-In: /run/systemd/system/snmpd.service.d
             └─override.conf
     Active: active (running) since Wed 2023-02-08 13:24:31 UTC; 1min 12s ago
    Process: 3918 ExecStartPre=/bin/mkdir -p /var/run/agentx (code=exited, status=0/SUCCESS)
   Main PID: 3919 (snmpd)
      Tasks: 1 (limit: 536)
     Memory: 7.5M
        CPU: 73ms
     CGroup: /system.slice/snmpd.service
             └─3919 /usr/sbin/snmpd -LS0-5d -Lf /dev/null -u Debian-snmp -g Debian-snmp -f -p /run/snmpd.pid

Feb 08 13:24:31 vyos systemd[1]: Starting Simple Network Management Protocol (SNMP) Daemon....
Feb 08 13:24:31 vyos systemd[1]: Started Simple Network Management Protocol (SNMP) Daemon..
vyos@vyos:~$ 

## Check snmpwalk from localhost
vyos@vyos:~$ snmpwalk -v 2c -c FOO localhost
SNMPv2-MIB::sysDescr.0 = STRING: VyOS 1.4-rolling-202302080317
SNMPv2-MIB::sysObjectID.0 = OID: SNMPv2-SMI::enterprises.44641
DISMAN-EVENT-MIB::sysUpTimeInstance = Timeticks: (17050) 0:02:50.50
SNMPv2-MIB::sysContact.0 = STRING: EMAIL
...
...
# Check restrictions when trying to get routing table:
vyos@vyos:~$ snmpwalk -v 2c -c FOO localhost .1.3.6.1.2.1.4.21
IP-MIB::ip.21 = No Such Object available on this agent at this OID

```
## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: data/templates/snmp/etc.snmpd.conf.j2 [hunk 1]
```diff
@@ -78,8 +78,8 @@ agentaddress unix:/run/snmpd.socket{{ ',' ~ options | join(',') if options is vy
 {%             endfor %}
 {%         endif %}
 {%         if comm_config.client is not vyos_defined and comm_config.network is not vyos_defined %}
-{{ comm_config.authorization }}community {{ comm }} -V RESTRICTED
-{{ comm_config.authorization }}community6 {{ comm }} -V RESTRICTED
+{{ comm_config.authorization }}community {{ comm }} 0.0.0.0/0 -V RESTRICTED
```
Comment (c-po) [inline data/templates/snmp/etc.snmpd.conf.j2]: This is an implicit default for the user and he will not know why. Please make this more transparent in using a `<defaultValue>` in the XML node, which will then also be added to the CLI help automatically. So the userbase will know that the network will default to 0.0.0.0/0 and ::/0.

**Review by c-po — CHANGES_REQUESTED** [review]

File: data/templates/snmp/etc.snmpd.conf.j2 (hunk 1 above)
Comment (sever-sever) [reply data/templates/snmp/etc.snmpd.conf.j2]: Is it a good idea to allow all addresses by default to get SNMP queries?
Or am I confusing something?

Comment (nicolas-fort) [reply data/templates/snmp/etc.snmpd.conf.j2]: IMO it's better to explicitly allow addresses.
This PR is just for backwards compatibility, so same configuration still works as it was working before latest changes in snmp.

But if we agree that it's better not to allow all access by default, we can close this PR, and prepare new updated and correct snmp documentation

Comment (c-po) [reply data/templates/snmp/etc.snmpd.conf.j2]: Moving this to `defaultValue` only adds "documentation" to the CLI and keeps the behavior "as is"

Comment (nicolas-fort) [reply data/templates/snmp/etc.snmpd.conf.j2]: In this case, defaultValue should set two default values: 0.0.0.0/0 and ::/0
Is this supported? As far as I know, it's not supported so far

Comment (c-po) [reply data/templates/snmp/etc.snmpd.conf.j2]: Using multiple defaults is supported, see commit 011e6a2cdea8fca73f7c433de5e2d6bc22bd9c19.

**Pull request merged**